src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -0,0 +1,274 @@
+        Utils.nonNull(pileup, "element list is null");
+        this.loc = loc;
+        this.pileupElements = pileup;
+    }
+
+    /**
+     * Create a new pileup without any aligned reads
+     */
+    public ReadPileup(final Locatable loc) {
+        this(loc, Collections.emptyList());
+    }
+
+    /**
+     * Create a new pileup with the given reads.
+     */
+    public ReadPileup(final Locatable loc, final List<GATKRead> reads, final int offset) {
+        this(loc, readsOffsetsToPileup(reads, offset));
+    }
+
+
+    /**
+     * Create a new pileup with the given reads.
+     */
+    public ReadPileup(final Locatable loc, final List<GATKRead> reads, final List<Integer> offsets) {
+        this(loc, readsOffsetsToPileup(reads, offsets));
+    }
+
+    /**
+     * Helper routine for converting reads and offset lists to a PileupElement list.
+     */
+    private static List<PileupElement> readsOffsetsToPileup(final List<GATKRead> reads, final List<Integer> offsets) {
+        Utils.nonNull(reads, "Illegal null read list in UnifiedReadBackedPileup");
+        Utils.nonNull(offsets, "Illegal null offsets list in UnifiedReadBackedPileup");
+        if (reads.size() != offsets.size()) {
+            throw new IllegalArgumentException("Reads and offset lists have different sizes!");
+        }
+
+        final List<PileupElement> pileup = new ArrayList<>(reads.size());
+        for (int i = 0; i < reads.size(); i++) {
+            pileup.add(PileupElement.createPileupForReadAndOffset(reads.get(i), offsets.get(i)));
+        }
+
+        return pileup;
+    }
+
+    /**
+     * Helper routine for converting reads and a single offset to a PileupElement list.
+     */
+    private static List<PileupElement> readsOffsetsToPileup(final List<GATKRead> reads, final int offset) {
+        Utils.nonNull(reads, "Illegal null read list");
+        return reads.stream().map(r -> PileupElement.createPileupForReadAndOffset(r, offset)).collect(Collectors.toList());
+    }
+
+    /**
+     * Make a new pileup consisting of elements of this pileup that satisfy the predicate.
+     * NOTE: the new pileup will not be independent of the old one (no deep copy of the underlying data is performed).
+     */
+    public ReadPileup makeFilteredPileup(final Predicate<PileupElement> filter){
+        return new ReadPileup(loc, pileupElements.stream().filter(filter).collect(Collectors.toList()));
+    }
+
+    /**
+     * Make a new pileup from elements whose reads have read groups that agree with the given lane ID.
+     * (if they have a name equal to the ID or starting with ID followed by a period ".").
+     * Also, if both laneID and read group ID are {@code null}, the read will be included.
+     * Returns empty pileup if no suitable elements are found.
+     * NOTE: the new pileup will not be independent of the old one (no deep copy of the underlying data is performed).
+     */
+    public ReadPileup getPileupForLane(final String laneID) {
```

Comment:
You've deleted a large number of methods from this class since the last review (eg., `getPileupForReadGroup()`, etc.). Were they truly unused (both now and in code that is likely to be ported soon?)